tesla_smart_charger/__main__.py

```
@@ -200,17 +222,41 @@ def set_config(config: Config) -> JSONResponse:
+    """Get the history of the charger."""
+    controller_db = None
+    try:
+        controller_db = db_controller.create_database_controller(
+            type, constants.DB_NAME, constants.DB_FILE_PATH
+        )
+        controller_db.initialize_db()
+    except Exception as e:
+        tsm_logger.error(f"Failed to initialize database: {e}")
+        raise HTTPException(
+            status_code=500, detail="Database connection not initialized"
+        )
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)